src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt

```diff
@@ -149,6 +174,50 @@ class RealHyperShard(
         return tests
     }
 
+    /**
+     * AST Java parsing to retrieve all the fully qualified test names in a file
+     *
+     * @param file a .java file
+     */
+    internal fun isJavaTest(file: File): Boolean {
+        checkIsJava(file)
+
+        var isTest = false
+        object : VoidVisitorAdapter<Any>() {
+            override fun visit(cu: CompilationUnit, arg: Any?) {
+                if (isTest) {
+                    return
+                }
+                super.visit(cu, arg)
+                val types = cu.types
+                for (type in types) {
+                    val classOrInterfaceDeclaration =
+                        type as? ClassOrInterfaceDeclaration ?: continue
+                    val shouldProcessTestMethod =
+                        shouldProcessTestMethods(classOrInterfaceDeclaration, annotationValue)
+                    if (shouldProcessTestMethod) {
+                        val methods = type.methods
+                        for (method in methods) {
+                            val annotationTest = method.getAnnotationByName("Test")
+                            if (annotationTest.isPresent) {
+                                isTest = true
```

Comment:
I could not find a good way to re-use this logic. I'm open to any ideas on this.